aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -1895,6 +1895,314 @@ mod tests {
+            .with_time_precision(TIME_PRECISION)
+            .build();
+
+        let leader_task = task.leader_view().unwrap();
+
+        let first_time = OLDEST_ALLOWED_REPORT_TIMESTAMP
+            .sub(&Duration::from_seconds(3 * TIME_PRECISION.as_seconds()))
+            .unwrap()
+            .to_batch_interval_start(&TIME_PRECISION)
+            .unwrap();
+        let first_batch_identifier =
+            TimeInterval::to_batch_identifier(&leader_task, &(), &first_time).unwrap();
+        let first_report_metadata = ReportMetadata::new(random(), first_time);
+
+        let second_time = OLDEST_ALLOWED_REPORT_TIMESTAMP
+            .add(&Duration::from_seconds(3 * TIME_PRECISION.as_seconds()))
+            .unwrap()
+            .to_batch_interval_start(&TIME_PRECISION)
+            .unwrap();
+        let second_batch_identifier =
+            TimeInterval::to_batch_identifier(&leader_task, &(), &second_time).unwrap();
+        let second_report_metadata = ReportMetadata::new(random(), second_time);
+
+        let verify_key: VerifyKey<VERIFY_KEY_LENGTH> = task.vdaf_verify_key().unwrap();
+
+        let first_transcript = run_vdaf(
+            vdaf.as_ref(),
+            verify_key.as_bytes(),
+            &(),
+            first_report_metadata.id(),
+            &0,
+        );
+        let second_transcript = run_vdaf(
+            vdaf.as_ref(),
+            verify_key.as_bytes(),
+            &(),
+            second_report_metadata.id(),
+            &0,
+        );
+
+        let agg_auth_token = task.aggregator_auth_token();
+        let helper_hpke_keypair = generate_test_hpke_config_and_private_key();
+        let first_report = generate_report::<VERIFY_KEY_LENGTH, Prio3Count>(
+            *task.id(),
+            first_report_metadata,
+            helper_hpke_keypair.config(),
+            first_transcript.public_share.clone(),
+            Vec::new(),
+            &first_transcript.leader_input_share,
+            &first_transcript.helper_input_share,
+        );
+        let second_report = generate_report::<VERIFY_KEY_LENGTH, Prio3Count>(
+            *task.id(),
+            second_report_metadata,
+            helper_hpke_keypair.config(),
+            second_transcript.public_share.clone(),
+            Vec::new(),
+            &second_transcript.leader_input_share,
+            &second_transcript.helper_input_share,
+        );
+
+        let aggregation_job_id = random();
+
+        let lease = ds
+            .run_unnamed_tx(|tx| {
+                let vdaf = Arc::clone(&vdaf);
+                let leader_task = leader_task.clone();
+                let first_report = first_report.clone();
+                let second_report = second_report.clone();
+
+                Box::pin(async move {
+                    tx.put_aggregator_task(&leader_task).await?;
+                    tx.put_client_report(vdaf.borrow(), &first_report).await?;
+                    tx.put_client_report(vdaf.borrow(), &second_report).await?;
```

Comment:
nit: I think it's better to call `unwrap()` here, because then if any of these three `put`s does fail, it's obvious which one it was. If we propagate the error then it blows up when we unwrap the result from `run_unnamed_tx` which is less helpful.

Reply:
Good call, I've been making this change as I touch code -- it is indeed much better to fail sooner rather than later in tests. I missed these as I adapted a lot of this code from a previously-existing test.